src/relay/pass/let_list.h

```diff
@@ -0,0 +1,75 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file let_list.h
+ * \brief Let List Let you Let stuff yo.
+ */
+#ifndef TVM_RELAY_PASS_LET_LIST_H_
+#define TVM_RELAY_PASS_LET_LIST_H_
+
+#include <utility>
+#include <vector>
+#include <tuple>
+#include "tvm/relay/type.h"
+#include "incomplete_type.h"
+
+namespace tvm {
+namespace relay {
+
+struct PartialVar {
+  Var v;
+  explicit PartialVar(const Var & v) : v(v) { }
+  PartialVar() : v(VarNode::make("x")) { }
+};
+
+struct LetList {
+  std::vector<std::tuple<PartialVar, Type, Expr>> lets;
```

Comment:
consider make it a struct, namedtuple is always better than unnamed version

Reply:
Do we really need the entry data structure, can we simply reuse LetNode while letting the body to be nullptr?